dist/d3-flamegraph.js

```diff
@@ -717,17 +728,11 @@ var flamegraph = function () {
     });
   }
 
-  function s4 () {
-    return Math.floor((1 + Math.random()) * 0x10000)
-      .toString(16)
-      .substring(1)
-  }
-
-  function injectIds (node) {
-    node.id = s4() + '-' + s4() + '-' + '-' + s4() + '-' + s4();
+  function injectIds (node, parent, pos) {
+    node.id = sha1((((pos || 0) + 1) ^ 3) * (node.depth ^ 7) + node.data.n + (parent || ''));
```

Comment:
Seems like there are Id clashes, or multiple nodes with the same id. Have you tried this solution with the default example (stacks.json)? There are missing frames, likely because of the duplicate ids.

Reply:
Thank you for your review. I apologize that I haven't check with the default example.

I will get it upstream and reconsider how to make it permanent but unique. My new strategy is simply to construct a string as an id inheriting parent id and combining that with its index instead of generating something non-self evident string. This should be able to avoid duplication when assuming that the graph is tree (which seems to be true).